src/format.rs

```diff
@@ -187,19 +187,57 @@ impl PayloadParams {
         let c1 = o.spec;
 
         if c0 == c1 {
+            // Exact match
             return Some(100);
-        } else {
-            // TODO: fuzzy matching.
         }
 
-        // No match
+        // Attempt fuzzy matching, since we don't have an exact match
+        if c0.codec != c1.codec || (c0.codec == Codec::Unknown && c1.codec == Codec::Unknown) {
```

Comment:
Should this `&&` be `||`?

Reply:
No I don't think so. We bail either if the codecs differ, but in the case that both codecs are unknown we also shouldn't treat that as a match, because we don't know that they are actually equal, only that str0m knows nothing about them